imqsauth/http.go

```diff
@@ -1115,72 +1105,26 @@ func getUserIdOrIdentity(r *httpRequest) (string, authaus.UserId, error) {
 	}
 }
 
-func (central *ImqsCentral) handleWebSocket(ws *websocket.Conn) {
-	closeConnection := true
-	var token authaus.Token
-	if token, err := authaus.HttpHandlerPrelude(&central.Config.Authaus.HTTP, central.Central, ws.Request()); err == nil {
-		if permList, errDecodePerms := authaus.PermitResolveToList(token.Permit.Roles, central.Central.GetRoleGroupDB()); errDecodePerms == nil {
-			closeConnection = !permList.Has(PermEnabled)
-		}
-	}
-
-	if closeConnection {
-		ws.Close()
-		return
+func (central *ImqsCentral) broadcastToAllSubscribers(msg string) {
+	notificationReqJson := &notificationRequestJson{}
+	notificationReqJson.Channel = "authNotifications"
+	notificationReqJson.Msg = msg
+	jsonString, err := json.Marshal(notificationReqJson)
+	if err != nil {
+		central.Central.Log.Errorf("Error notifying distributor about auth group changes: %v", err)
 	}
 
-	id := central.addSubscriber(ws)
-
-	central.Central.Log.Infof("Added auth web socket client %v : %v", id, token.Identity)
-	var msg string
-	for {
-		// We want to timeously close and remove the web sockets that send
-		// EOF's or invalid data, not wait until we fail to send and only then close them
-		// We just ignore whatever else we receive
-		if err := websocket.Message.Receive(ws, &msg); err != nil {
-			closeAndDeleteWebSocket(central, id)
-			return
-		}
-	}
-}
-
-func closeAndDeleteWebSocket(central *ImqsCentral, id uint64) {
-	central.subscriberLock.Lock()
-	defer central.subscriberLock.Unlock()
-
-	ws, ok := central.roleChangeSubscribers[id]
-	if ok {
-		ws.Close()
+	req, err := http.NewRequest("POST", central.Config.NotificationUrl, bytes.NewReader(jsonString))
+	if err != nil {
+		central.Central.Log.Errorf("Error notifying distributor about auth group changes: %v", err)
 	}
-	delete(central.roleChangeSubscribers, id)
-}
-
-func (central *ImqsCentral) broadcastToAllSubscribers(msg string) {
-	var deleteIDs []uint64
-	central.subscriberLock.RLock()
-
-	for id, ws := range central.roleChangeSubscribers {
-		err := websocket.Message.Send(ws, msg)
-		if err != nil {
-			// only record the id, since we already have a read lock active at this point
-			deleteIDs = append(deleteIDs, id)
-		}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		central.Central.Log.Errorf("Error notifying distributor about auth group changes: %v", err)
 	}
-
-	central.subscriberLock.RUnlock()
-
-	// now delete the recorded id's
-	for _, id := range deleteIDs {
-		closeAndDeleteWebSocket(central, id)
+	defer resp.Body.Close()
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		central.Central.Log.Errorf("Error notifying distributor about auth group changes: %v\n%v", err, string(respBody))
```

Comment:
This falls inline with the other story I just did where devops requested we change as many Errorf statements to Warnf statements so that loggly does not report on everything, can we change these?